gsl/span

```diff
@@ -472,13 +472,13 @@ public:
 
     constexpr span<element_type, dynamic_extent> first(index_type count) const
     {
-        Expects(count >= 0 && count <= size());
+        Expects(count >= 0 && (count < size() || count == size()));
         return {data(), count};
```

Comment:
**if** you are going to call `size()` more than once, you should store the result in a local variable and reuse it.
Another occurrence below.